core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/auth/BaseGroupsRule.scala

```diff
@@ -58,8 +60,8 @@ abstract class BaseGroupsRule(val settings: Settings)
         resolveGroupIds(blockContext) match {
           case None =>
             Task.now(Rejected())
-          case Some(resolvedPermittedGroupIds) if blockContext.isCurrentGroupEligible(resolvedPermittedGroupIds) =>
-            continueCheckingWithUserDefinitions(blockContext, resolvedPermittedGroupIds)
+          case Some(groupIds) if blockContext.isCurrentGroupEligible(groupsPotentiallyPermittedByRule(groupIds)) =>
```

Comment:
The previous name was a little bit better. In this class, groups can have several contexts: groups from the settings, groups from the rule, groups from the external system, and resolved groups (and maybe even more). Maybe, we should have more types to distinguish them. Anyway, when we call it `groupIds` we miss the information about the mentioned context.

The comment applies to other places like this in this file.